clustering/web/infinispan/src/test/java/org/wildfly/clustering/web/infinispan/routing/RankedRouteLocatorTestCase.java

```diff
@@ -0,0 +1,158 @@
+        LocalizedCacheTopology localizedTopology = new LocalizedCacheTopology(mode, topology, this.partitioner, manager.getAddress(), true);
+        when(this.dist.getCacheTopology()).thenReturn(localizedTopology);
+    }
+
+    @SuppressWarnings({ "unchecked", "rawtypes" })
+    @Test
+    public void test() {
+        RankedRouteLocatorConfiguration config = mock(RankedRouteLocatorConfiguration.class);
+
+        when(config.getCache()).thenReturn((AdvancedCache) this.cache);
+        when(config.getMemberFactory()).thenReturn(this.factory);
+        when(config.getRegistry()).thenReturn(this.registry);
+        when(config.getDelimiter()).thenReturn(".");
+        when(config.getMaxRoutes()).thenReturn(3);
+        for (int i = 0; i < this.addresses.length; ++i) {
+            when(this.factory.createNode(this.addresses[i])).thenReturn(this.members[i]);
+            when(this.registry.getEntry(this.members[i])).thenReturn(new AbstractMap.SimpleImmutableEntry<>(String.valueOf(i), null));
+        }
+        when(this.registry.getGroup()).thenReturn(this.group);
+        when(this.group.getLocalMember()).thenReturn(this.localMember);
+        when(this.registry.getEntry(this.localMember)).thenReturn(new AbstractMap.SimpleImmutableEntry<>("local", null));
+
+        RouteLocator locator = new RankedRouteLocator(config);
+
+        switch (this.cache.getCacheConfiguration().clustering().cacheMode()) {
+            case DIST_SYNC: {
+                when(this.partitioner.getSegment(new Key<>("session"))).thenReturn(0);
+                String result = locator.locate("session");
+                Assert.assertEquals("0.1.local", result);
+
+                when(this.partitioner.getSegment(new Key<>("session"))).thenReturn(1);
+                result = locator.locate("session");
+                Assert.assertEquals("1.2.local", result);
+
+                when(this.partitioner.getSegment(new Key<>("session"))).thenReturn(2);
+                result = locator.locate("session");
+                Assert.assertEquals("2.0.local", result);
+                break;
+            }
+            case REPL_SYNC: {
+                when(this.partitioner.getSegment(new Key<>("session"))).thenReturn(0);
+                String result = locator.locate("session");
+                Assert.assertEquals("0.1.2", result);
+
+                when(this.partitioner.getSegment(new Key<>("session"))).thenReturn(1);
+                result = locator.locate("session");
+                Assert.assertEquals("1.2.0", result);
+
+                when(this.partitioner.getSegment(new Key<>("session"))).thenReturn(2);
+                result = locator.locate("session");
+                Assert.assertEquals("2.0.1", result);
+                break;
+            }
+            default: {
+                String result = locator.locate("session");
+                Assert.assertEquals("local", result);
+            }
+        }
+    }
+}
```

Comment:
@pferraro could you give me a quick explanation about this test? I have difficulties understanding what it exactly does

Reply:
@tommaso-borgato It validates that the RouteLocator implementation returns the expected results from a given cache topology for a given cache mode.

Reply:
@pferraro is it correct to say that it tests that `JVM route` portion of the JSESSIONID cookie is correctly filled in based on the cache topology?

Reply:
Yes.